### PR TITLE
net/freeradius: Add basic LDAP support to FreeRADIUS

### PR DESCRIPTION
**Pull request by mimugmail**
This update adds basic LDAP support. It allows to bind via LDAP and LDAPS, certification validation is disabled ATM, will follow after 18.1 or so. Also there's no search filter option available yet, but I need a devel pkg for the community to test. Perhaps this can go stable with 18.1, depends on the guys in the forums :)

### CODE REVIEW

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/Api/LdapController.php [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ *
+ */
+
+namespace OPNsense\Freeradius\Api;
+
+use \OPNsense\Base\ApiControllerBase;
+use \OPNsense\Freeradius\Ldap;
+use \OPNsense\Core\Config;
+
+class LdapController extends ApiControllerBase
```
Comment (fichtner) [inline net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/Api/LdapController.php]: I think the Mutable controller should be used if you can from now on, see https://github.com/opnsense/plugins/pull/442

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml [hunk 2]
```diff
@@ -11,6 +11,12 @@
         <type>checkbox</type>
         <help>This allows you to dynamically assign VLANs on your physical switch ports.</help>
     </field>
+    <field>
+        <id>general.ldap_enabled</id>
+        <label>Enable LDAP</label>
+        <type>checkbox</type>
+        <help>This allows you to bind to an external LDAP server, use configuration in submenu LDAP.</help>
```
Comment (fichtner) [inline net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml]: Remove second half of the text, it embeds information that can and will change long-term.

**Comment by mimugmail** [comment]
Seems I'm getting sick oO
Now it's fixed ..

File: net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml [hunk 3]
```diff
@@ -15,7 +15,7 @@
         <id>general.ldap_enabled</id>
         <label>Enable LDAP</label>
         <type>checkbox</type>
-        <help>This allows you to bind to an external LDAP server, use configuration in submenu LDAP.</help>
+        <help>This allows you to bind to an external LDAP server. Be aware that FreeRADIUS wont start if there's no further configuration.</help>
```
Comment (fichtner) [inline net/freeradius/src/opnsense/mvc/app/controllers/OPNsense/Freeradius/forms/general.xml]: will not
there is
:)

**Comment by mimugmail** [comment]
Jetzt aber ... :)

**Comment by fichtner** [comment]
Dankeschön!

**Pull request merged**